gateway/mw_redis_cache.go

```diff
@@ -137,26 +152,22 @@ func (m *RedisCacheMiddleware) decodePayload(payload string) (string, string, er
 
 // ProcessRequest will run any checks on the request on the way through the system, return an error to have the chain fail
 func (m *RedisCacheMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Request, _ interface{}) (error, int) {
-	// Only allow idempotent (safe) methods
-	if r.Method != "GET" && r.Method != "HEAD" && r.Method != "OPTIONS" && r.Method != "POST" {
-		return nil, http.StatusOK
-	}
-
 	var stat RequestStatus
 	var cacheKeyRegex string
+	var cacheMeta *EndPointCacheMeta
 
 	_, versionPaths, _, _ := m.Spec.Version(r)
 	isVirtual, _ := m.Spec.CheckSpecMatchesStatus(r, versionPaths, VirtualPath)
 
 	// Lets see if we can throw a sledgehammer at this
-	if m.Spec.CacheOptions.CacheAllSafeRequests && r.Method != "POST" {
+	if m.Spec.CacheOptions.CacheAllSafeRequests && isSafeMethod(r.Method) {
```

Comment:
This code should run only in case of safe method.
Previously, only safe methods and POST could get to this code.
Since I opened the cache for all of the methods the condition of `!= "POST"` is not good enough and we have to add this logic.